pkg/operator/daemonset.go

```diff
@@ -80,6 +80,9 @@ func MakeDaemonSet(fb fluentbitv1alpha2.FluentBit, logPath string) appsv1.Daemon
 							},
 							ReadinessProbe: fb.Spec.ReadinessProbe,
 							LivenessProbe:  fb.Spec.LivenessProbe,
+							SecurityContext: &corev1.SecurityContext{
+								Privileged: func(b bool) *bool { return &b }(true),
+							},
```

Comment:
@wanjunlei I think we needn't make fluentbit pod privileged pod to read logs from k8s node?

Reply:
Yes, the fluentbit pod should not be a privileged pod by default.

If you want to make fluentbit pod a privileged pod, please set a `SecurityContext` in [FlunetBit CRD](https://github.com/fluent/fluent-operator/blob/master/apis/fluentbit/v1alpha2/fluentbit_types.go#L73).

Reply:
I'll test this tomorrow to figure out what OpenShift was preventing the unprivileged pod from doing.

Reply:
> If you want to make fluentbit pod a privileged pod, please set a `SecurityContext` in [FlunetBit CRD](https://github.com/fluent/fluent-operator/blob/master/apis/fluentbit/v1alpha2/fluentbit_types.go#L73).

Right, then @jseutter I think you can revert this change to the default of setting securityContext to true.
You can set it in the FluentBit CRD if you need it

Reply:
Hm, I don't see a way to set fluentbit pods to run as a privileged user.  The only privileged section in the FluentBit CRD is under spec.initContainers.securityContext which won't help me.  Is there a reason there is no spec.securityContext.privileged attribute?

Reply:
You need to add a `SecurityContext` field into `Fluentd` CRD like [this](https://github.com/fluent/fluent-operator/blob/master/apis/fluentbit/v1alpha2/fluentbit_types.go#L73).

Reply:
Hm, I'm talking about FluentBit in this case, not Fluentd.  Regardless, I modified the FluentBit CRD to allow spec.securityContext.privileged but the operator code won't include that setting when creating the StatefulSet.  I'll dig into the operator code next.

Reply:
Sorry, my mistake.

The StatefulSet is controlled by the [Collector](https://github.com/fluent/fluent-operator/blob/master/apis/fluentbit/v1alpha2/collector_types.go#L31) CRD, not the `FluentBit` CRD.

Reply:
I think I figured it out.  FluentBit links to a PodSecurityContext which does not have a privileged field.  There is a SecurityContext which does have a privileged field, and is used for container security contexts.  I figured out a workaround which is to use the seLinuxOptions and set the security context:

`securityContext:
  seLInuxOptions:
    level: s0
    role: system_r
    type: spc_t
    user: system_u`

Doing this it doesn't matter if the container user does not have access to the files.  I have removed the offending code from the PR.